sympy/physics/mechanics/kane.py

```diff
@@ -286,13 +298,21 @@ def _initialize_kindiffeq_matrices(self, kdeqs):
             #
             # k_kqdot^-1*k_ku*u(t) + I*q'(t) + k_kqdot^-1*f_k = 0
             #
-            # NOTE : Solving the kinematic differential equations here is not
-            # necessary and prevents the equations from being provided in fully
-            # implicit form.
-            f_k = k_kqdot.LUsolve(f_k)
-            k_ku = k_kqdot.LUsolve(k_ku)
-            k_kqdot = eye(len(qdot))
-            self._qdot_u_map = dict(zip(qdot, -(k_ku*u + f_k)))
+            if explicit_kinematics:
+                # NOTE : Solving the kinematic differential equations here is not
+                # necessary and prevents the equations from being provided in fully
+                # implicit form.
+                f_k = k_kqdot.LUsolve(f_k)
+                k_ku = k_kqdot.LUsolve(k_ku)
+                k_kqdot = eye(len(qdot))
+                self._qdot_u_map = dict(zip(qdot, -(k_ku*u + f_k)))
+            else:
+                # If kinematics are not forced to be explicit, do not provide a qdot_u_map
+                # However, still compute a kindiffdict that is necessary to provide a RHS
+                # (i.e. fully explicit) version of the equations
+                self._qdot_u_map = dict()
```

Comment:
If you don't calculate this here, then want the equations of motion be incorrectly formed. The qdots have to be replaced by expressions of the u's before partial velocities are computed, no?